pkg/nodebootstrap/userdata.go

```diff
@@ -110,12 +110,38 @@ func makeKubeletConfigYAML(spec *api.ClusterConfig, ng *api.NodeGroup) ([]byte,
 
 	// Set default reservations if specs about instance is available
 	if info, ok := instanceTypeInfos[ng.InstanceType]; ok {
+		// This is a NodeGroup with a single instanceType defined
 		if _, ok := obj["kubeReserved"]; !ok {
 			obj["kubeReserved"] = api.InlineDocument{}
 		}
 		obj["kubeReserved"].(api.InlineDocument)["ephemeral-storage"] = info.DefaultStorageToReserve()
 		obj["kubeReserved"].(api.InlineDocument)["cpu"] = info.DefaultCPUToReserve()
 		obj["kubeReserved"].(api.InlineDocument)["memory"] = info.DefaultMemoryToReserve()
```

Comment:
👍  I created the new function but the signature is different than suggested. Rather than passing in the `kubeReserved` document and mutating it, I pass only the `info` and return the required document. Let me know if this works!